### PR TITLE
Morphing

### PR DESCRIPTION
**Pull request by guycalledfrank**
Work in progress.

### CODE REVIEW

File: src/resources/parser/json-model.js [hunk 1]
```diff
@@ -333,6 +406,14 @@ pc.extend(pc, function () {
                     meshInstance.skinInstance = skinInstances[skinIndex];
                 }
 
+                if (mesh.morph) {
+                    var morphIndex = morphs.indexOf(mesh.morph);
+                    if (morphIndex === -1) {
+                        throw new Error('Mesh\'s morph does not appear in morph array.');
```
Comment (Maksims) [inline src/resources/parser/json-model.js]: Do not throw exceptions from engine, unless in DEBUG mode.

File: src/scene/forward-renderer.js [hunk 2]
```diff
@@ -1353,6 +1354,47 @@ pc.extend(pc, function () {
...
+            // #ifdef PROFILER
+            var morphTime = pc.now();
+            // #endif
+
+            var i, morph;
+            var drawCallsCount = drawCalls.length;
+            for (i = 0; i < drawCallsCount; i++) {
+                morph = drawCalls[i].morphInstance;
+                if (morph) {
+                    if (morph._dirty) {
```
Comment (Maksims) [inline src/scene/forward-renderer.js]: Why not:
`if (morph && morph._dirty) {`
?

File: src/resources/parser/json-model.js (hunk 1 above)
Comment (guycalledfrank) [inline src/resources/parser/json-model.js]: This is basically a copy-pasted code from skinning parser. Should change both then.

File: src/scene/morph.js [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+        this.name = options.name;
+    };
+
+    /**
+     * @name pc.Morph
+     * @class Contains a list of pc.MorphTarget, a combined AABB and some associated data.
+     * @param {pc.MoprhTarget[]} targets A list of morph targets
+     */
+    var Morph = function (targets) {
+        this.aabb = new pc.BoundingBox(new pc.Vec3(), new pc.Vec3());
```
Comment (Maksims) [inline src/scene/morph.js]: If not vectors are provided it will allocate in constructor.

**Pull request merged**